expense/src/workflows/expense.ts

```diff
@@ -0,0 +1,43 @@
+import { createExpense } from '@activities/createExpense';
+import { payment } from '@activities/payment' ;
+import { Trigger, sleep } from '@temporalio/workflow';
+
+type expenseStatus = 'CREATED' | 'APPROVED' | 'REJECTED' | 'TIMED_OUT' | 'COMPLETED';
+let status: expenseStatus = 'CREATED';
+
+const signalTrigger = new Trigger<void>();
+const timeoutMS = 10000;
+
+const signals = {
+  approve() {
+    status = 'APPROVED';
+    signalTrigger.resolve();
+  },
+  reject() {
+    status = 'REJECTED';
+    signalTrigger.resolve();
+  }
+}
+
+async function main(expenseId: string): Promise<{ status: string }> {
+  await createExpense(expenseId);
+
+  if (status === 'CREATED') {
+    await Promise.race([signalTrigger, sleep(timeoutMS)]);
```

Comment:
I don't want to recommend doing that, because it might cause a race condition. `Promise.race()` doesn't cancel promises, so the `then(() => { status = 'TIMED_OUT' })` block would always execute. With the current implementation, that wouldn't be a problem, but you could see a case where that might lead to `status` getting updated in an unexpected way.

Reply:
Yeah, very good point, even though there's nothing async going on here until we `await payment()` below.
May I suggest you make the trigger type `Trigger<ExpenseStatus>` and resolve it with `accepted` or `rejected`.

```ts
status = await Promise.race([signalTrigger, sleep(timeoutMS).then(() => 'TIMED_OUT')]);
```

Reply:
I like your suggestion, I think that's the right way to do this. Just implemented :+1: